osbrain/tests/test_agent.py

```diff
@@ -50,7 +50,9 @@ def test_early_agent_proxy(nsaddr):
     # Locate agent now
     a0 = Proxy('a0', timeout=3.)
     # Just check agent is ready
-    assert a0.ping() == 'pong'
+    safe_status = a0._safe
+    assert a0.unsafe.ping() == 'pong'
+    a0._safe = safe_status
```

Comment:
I answer here so as to not duplicate answers. All tests pass if I remove the "store status" lines, but I think it might not be doing what we want, since the status is then kept even after the call (even though this may be what we want, I'm not sure). Here is a reduced example:

```
class C():
  def __init__(self):
    self.x = 3
    
  @property
  def set_four(self):
    self.x = 4
    return self
    
  def print_x(self):
    print(self.x)
    
c = C()
c.print_x() # prints 3
c.set_four.print_x() # prints 4
c.print_x() # prints 4
```

Since before we were calling ready and ping separately, there was no status modification, but now there is. Perhaps we could create a method for specifically unsafe calling a method, despite configuration being set to safe?